DEC/src/roe/RoeUtility.jl

```diff
@@ -0,0 +1,222 @@
+# """
+# Defines Roe, a struct which acts as a wrapper for e-graph typed in the Sorts of a given theory, as well as functions for manipulating it.
+# """
+# module RoeUtility
+
+# using ..SSAExtract: SSA, SSAVar, SSAExpr, extract_ssa!
+using ..Util.HashColor
+
+using StructEquality
+import Metatheory
+using Metatheory: EGraph, EGraphs, Id, VECEXPR_FLAG_ISCALL, VECEXPR_FLAG_ISTREE, VECEXPR_META_LENGTH
+using MLStyle
+using Reexport
+
+"""
+Sorts in each theory are subtypes of this abstract type.
+"""
+abstract type AbstractSort end
+export AbstractSort
+
+"""    TypedApplication
+
+Struct containing a Function and the vector of Sorts it requires.
+"""
+@struct_hash_equal struct TypedApplication
```

Comment:
This, and other things that have AbstractSort parameters, should be typed based on a subtype of AbstractSort.